### ISSUE TITLE
New browser window opened when no command was issued

### ISSUE THREAD

**Issue by codecat555** [talonvoice/talon#300]
I just saw something odd. I decided to document it here, in case it happens again and I want to chase it down. I don't know if it was actually a talon issue or not. 

I was browsing with firefox on windows. I have vimium loaded, so I suppose it could have been a bug in that software.

Anyways, I had wheel gaze active and had just used middle click to open a link in a new tab. I scrolled down a few more pages and then  my wheel gaze was canceled, due to an unintended pop (I was actually greeting my cat with a little clicking sound I make). 

I decided to disable the `user.mouse_enable_pop_stops_scroll` setting, so I switched to a wsl window and edited my settings file. Upon save, the talon console showed the file was reloaded without errors.

Then, when I switched the focus back to firefox, and was surprised when a new browser window popped open to show the same page I had previously opened in a new tab. The talon console didn't show that any voice commands were detected and, in fact, I had not made any sounds.

Both my hands were resting in my lap at this time, so it wasn't due to bumping the mouse or something like that.

FWIW, here's the relevant part of the log. It begins with the mid-click used to open the link in a new tab.

Also - 

`app switch : key(ctrl-alt-tab)`
`clap off: mimic("sleep all")`
`win back: key(alt-tab enter)`

```
2021-05-04 09:53:33    IO EMIT   mini
2021-05-04 09:53:33    IO DECODE ['mitty']
2021-05-04 09:53:33    IO [audio]=810.000ms  [compile]=0.000ms (0.00x)  [flac]=0.000ms (0.00x)  [request]=177.821ms (0.22x)  [emit]=132.575ms (0.16x)  [decode]=2.604ms (0.00x)  [total]=313.000ms (0.17x)
2021-05-04 09:53:33    IO
2021-05-04 09:54:17    IO EMIT   lap switch
2021-05-04 09:54:17    IO DECODE ['app', 'switch']
2021-05-04 09:54:17    IO [audio]=1170.000ms  [compile]=0.000ms (0.00x)  [flac]=15.000ms (0.01x)  [request]=253.930ms (0.22x)  [emit]=149.248ms (0.13x)  [decode]=2.822ms (0.00x)  [total]=421.000ms (0.13x)
2021-05-04 09:54:17    IO
2021-05-04 09:54:20    IO COMPILING
sum=187.273ms load=0ms build=15.643ms link=0ms flat=0ms min=171.63ms cfg=0ms
2021-05-04 09:54:20    IO grammar -> dfa -> cfg in 0.188000s
2021-05-04 09:54:20    IO EMIT   go down
2021-05-04 09:54:20    IO DECODE ['go', 'down']
2021-05-04 09:54:20    IO [audio]=990.000ms  [compile]=188.000ms (0.19x)  [flac]=0.000ms (0.00x)  [request]=213.745ms (0.22x)  [emit]=143.261ms (0.14x)  [decode]=1.994ms (0.00x)  [total]=547.000ms (0.15x)
2021-05-04 09:54:20    IO
2021-05-04 09:54:22    IO EMIT   and to
2021-05-04 09:54:22    IO DECODE ['anter']
2021-05-04 09:54:22    IO [audio]=990.000ms  [compile]=0.000ms (0.00x)  [flac]=15.000ms (0.02x)  [request]=204.830ms (0.21x)  [emit]=134.611ms (0.14x)  [decode]=4.560ms (0.00x)  [total]=359.000ms (0.14x)
2021-05-04 09:54:22    IO
2021-05-04 09:54:27    IO COMPILING
sum=169.367ms load=6.508ms build=0ms link=0ms flat=0ms min=162.859ms cfg=0ms
2021-05-04 09:54:27    IO grammar -> dfa -> cfg in 0.187000s
2021-05-04 09:54:27    IO EMIT   v i
2021-05-04 09:54:27    IO DECODE ['v', 'i']
2021-05-04 09:54:27    IO [audio]=1020.000ms  [compile]=187.000ms (0.18x)  [flac]=0.000ms (0.00x)  [request]=211.548ms (0.21x)  [emit]=129.189ms (0.13x)  [decode]=3.263ms (0.00x)  [total]=531.000ms (0.13x)
2021-05-04 09:54:27    IO
2021-05-04 09:54:29    IO EMIT   settings
2021-05-04 09:54:29    IO DECODE ['string', 'i']
2021-05-04 09:54:29    IO [audio]=1020.000ms  [compile]=0.000ms (0.00x)  [flac]=0.000ms (0.00x)  [request]=224.642ms (0.22x)  [emit]=132.764ms (0.13x)  [decode]=2.594ms (0.00x)  [total]=360.000ms (0.13x)
2021-05-04 09:54:29    IO

... [editing settings.talon]...

2021-05-04 09:55:19 DEBUG [-] C:\Users\me\AppData\Roaming\talon\user\personal\settings.talon
2021-05-04 09:55:19 DEBUG [+] C:\Users\me\AppData\Roaming\talon\user\personal\settings.talon
2021-05-04 09:55:23    IO COMPILING
sum=222.803ms load=0ms build=53.464ms link=15.595ms flat=0ms min=153.744ms cfg=0ms
2021-05-04 09:55:23    IO grammar -> dfa -> cfg in 0.234000s
2021-05-04 09:55:24    IO EMIT   ab switch
2021-05-04 09:55:24    IO DECODE ['app', 'switch']
2021-05-04 09:55:24    IO [audio]=1080.000ms  [compile]=234.000ms (0.22x)  [flac]=16.000ms (0.01x)  [request]=224.730ms (0.21x)  [emit]=148.773ms (0.14x)  [decode]=1.497ms (0.00x)  [total]=625.000ms (0.14x)
2021-05-04 09:55:24    IO
2021-05-04 09:55:25    IO COMPILING
sum=169.705ms load=0ms build=15.628ms link=0ms flat=0ms min=154.077ms cfg=0ms
2021-05-04 09:55:25    IO grammar -> dfa -> cfg in 0.172000s
2021-05-04 09:55:26    IO EMIT   escape
2021-05-04 09:55:26    IO DECODE ['escape']
2021-05-04 09:55:26    IO [audio]=900.000ms  [compile]=172.000ms (0.19x)  [flac]=16.000ms (0.02x)  [request]=198.124ms (0.22x)  [emit]=129.349ms (0.14x)  [decode]=0.526ms (0.00x)  [total]=516.000ms (0.14x)
2021-05-04 09:55:26    IO
2021-05-04 09:55:27    IO COMPILING
sum=169.357ms load=15.628ms build=0ms link=0ms flat=0ms min=153.729ms cfg=0ms
2021-05-04 09:55:27    IO grammar -> dfa -> cfg in 0.172000s
2021-05-04 09:55:28    IO EMIT   win back
2021-05-04 09:55:28    IO DECODE ['win', 'back']
2021-05-04 09:55:28    IO [audio]=1080.000ms  [compile]=172.000ms (0.16x)  [flac]=0.000ms (0.00x)  [request]=179.030ms (0.17x)  [emit]=132.903ms (0.12x)  [decode]=1.066ms (0.00x)  [total]=485.000ms (0.12x)
2021-05-04 09:55:28    IO
2021-05-04 09:55:35    IO COMPILING
sum=200.621ms load=0ms build=15.657ms link=0ms flat=0ms min=169.375ms cfg=15.589ms
2021-05-04 09:55:36    IO grammar -> dfa -> cfg in 0.203000s
2021-05-04 09:55:36    IO EMIT   widen clothes
2021-05-04 09:55:36    IO DECODE ['window', 'close']
2021-05-04 09:55:36    IO [audio]=1080.000ms  [compile]=203.000ms (0.19x)  [flac]=0.000ms (0.00x)  [request]=191.976ms (0.18x)  [emit]=130.483ms (0.12x)  [decode]=5.541ms (0.01x)  [total]=531.000ms (0.13x)
2021-05-04 09:55:36    IO
2021-05-04 09:56:07    IO EMIT   clap off
2021-05-04 09:56:07    IO DECODE ['clap', 'off']
2021-05-04 09:56:07    IO [audio]=960.000ms  [compile]=0.000ms (0.00x)  [flac]=0.000ms (0.00x)  [request]=221.413ms (0.23x)  [emit]=135.355ms (0.14x)  [decode]=2.233ms (0.00x)  [total]=359.000ms (0.14x)
2021-05-04 09:56:07    IO
2021-05-04 09:56:11    IO COMPILING
sum=15.635ms load=15.635ms build=0ms link=0ms flat=0ms min=0ms cfg=0ms
2021-05-04 09:56:11    IO grammar -> dfa -> cfg in 0.015000s

```

**Comment by lunixbochs** [comment]
If this sort of thing happens again, post events.tail() from the repl

**Comment by codecat555** [comment]
Thanks for mentioning that. I was able to capture the event log using `events.tail(limit=0)`. I found the relevant section and will upload it.

I believe the answer is to be found on line 991 - where the 'win back' command I have defined presses the 'enter' key.

I guess the 'alt-tab' part alone is sufficient for my purposes (going back to the previous window)...although I have been using that command as-is for a week or so now with no problems. The 'enter' would only be required if I used 'ctrl-alt-tab' instead.

Further, after discovering that, I repeated the steps and found that the 'enter' did in fact trigger firefox to open the currently-selected link (which was the last one clicked on that page)...BUT, in that case it opened the link in a new tab rather than a new window.

Anyways, I'm satisfied with the outcome at this point. This was a worthwhile exercise for me to learn how to capture debug info and read the events log. Thanks for your help!

[issue300-events.log](https://github.com/talonvoice/talon/files/6423412/issue300-events.log)
